modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/DefaultApplicationFactory.java

```diff
@@ -126,6 +127,11 @@ public Application createArtifact(ApplicationDescriptor descriptor) throws IOExc
         pluginDependenciesResolver.resolve(domain.getDescriptor().getPlugins(),
                                            new ArrayList<>(getArtifactPluginDescriptors(descriptor)));
 
+    // Refreshes the list of plugins on the descriptor with the resolved from domain and transitive plugin dependencies
+    Set resolvedArtifactPlugins = new LinkedHashSet<>();
+    resolvedArtifactPlugins.addAll(resolvedArtifactPluginDescriptors);
+    descriptor.setPlugins(resolvedArtifactPlugins);
```

Comment:
I don't like this as it is modifying the descriptor, other approach would be to have a resolvedPlugins field and set the list of plugins descriptor as they were resolved (using semver and also depending on what are the plugins already defined at the domain level)

Reply:
Should at least log a warning if the transitive dependencies versions are inconsistent, IMO

Reply:
Something like: